lib/event-jobs.js

```diff
@@ -0,0 +1,125 @@
+  this._subscribedEvents = opts.subscribedEvents || [];
+
+  this._name = opts.name;
+  if (!this._name) {
+    debug('error: name required for EventJobs');
+    throw new Error('name required for EventJobs');
+  }
+
+  this._channel = opts.channel;
+  if (!this._channel) {
+    debug('error: channel required for EventJobs');
+    throw new Error('channel required for EventJobs');
+  }
+}
+
+/**
+ * checks to see if event is a publish event
+ * @param  {String}  eventName name of exchange to publish to
+ */
+EventJobs.prototype.isPublishEvent = function (eventName) {
+  debug('isPublishEvent', eventName);
+  return ~this._publishedEvents.indexOf(eventName);
+};
+
+/**
+ * creates an exchanges for each events we want to publish
+ * @param  {Function} cb (err)
+ */
+EventJobs.prototype.assertExchanges = function (cb) {
+  debug('assertExchanges');
+  async.forEach(this._publishedEvents, this._createExchange.bind(this), cb);
+};
+
+/**
+ * creates an exchange
+ * @see http://www.squaremobius.net/amqp.node/channel_api.html#channel_assertExchange
+ * @param  {Function} cb (err)
+ */
+EventJobs.prototype._createExchange = function (exchangeName, cb) {
+  debug('_createExchange', exchangeName);
+  this._channel.assertExchange(exchangeName, 'fanout', { durable: true }, cb);
+};
+
+/**
+ * used to publish events to a certain exchange
+ * do nothing if event is not one defined in _publishedEvents
+ * @see http://www.squaremobius.net/amqp.node/channel_api.html#channel_publish
+ * @param  {String}   eventName name of exchange to publish too
+ * @param  {Object}   data      job data
+ */
+EventJobs.prototype.publish = function (eventName, data) {
+  debug('publish', eventName, data);
+  this._channel.publish(eventName, '', data);
+};
+
+/**
+ * checks to see if event is part of _subscribedEvents
+ * @param  {String}  eventName name of queue to subscribe to
+ */
+EventJobs.prototype.isSubscribeEvent = function (eventName) {
+  debug('isSubscribeEvent', eventName);
+  return ~this._subscribedEvents.indexOf(eventName);
+};
+
+/**
+ * setup queues passed for each event we want to listen for
+ * @param  {Function} cb (err)
+ */
+EventJobs.prototype.createQueues = function (cb) {
+  debug('createQueues');
+  async.forEach(this._subscribedEvents, this._createQueue.bind(this), cb);
+};
+
+/**
+ * asserts a queue and then binds that to the event exchange
+ * appends name to queue to have a predictable queue name
+ * @see http://www.squaremobius.net/amqp.node/channel_api.html#channel_assertQueue
+ * @see http://www.squaremobius.net/amqp.node/channel_api.html#channel_bindQueue
+ * @param  {String}   eventName name to create queue for
+ * @param  {Function} cb    (err)
+ */
+EventJobs.prototype._createQueue = function (eventName, cb) {
+  debug('_createQueue', eventName);
+  var self = this;
+
+  var queueName = [self._name, eventName, 'queue'].join('.');
+  self._channel.assertQueue(queueName, { durable: true }, function (err) {
```

Comment:
interesting part from the book:

>>A fanout exchange type ensures every bound queue gets a copy of every message by ignoring any routing key that you specify in the binding or message. To prevent messages from piling up in queues endlessly, an exclusive and auto-deleted queue should be used for each subscriber. This allows a subscriber to receive messages when it is connected, and have the queue deleted when the subscriber disconnects.


Queue should be `exclusive` and `auto-deleted`.  What do you think about this @anandkumarpatel?